bullet_train-fields/app/javascript/controllers/fields/super_select_controller.js

```diff
@@ -1,161 +1,182 @@
 import { Controller } from "@hotwired/stimulus"
-require("select2/dist/css/select2.min.css");
-import select2 from "select2";
-
-const select2SelectedPreviewSelector = ".select2-selection--single"
-const select2SearchInputFieldSelector = ".select2-search__field"
+require("tom-select/dist/css/tom-select.css")
+import TomSelect from "tom-select/dist/js/tom-select.complete.js"
 
 export default class extends Controller {
   static targets = [ "select" ]
-  static values = {
-    acceptsNew: Boolean,
-    enableSearch: Boolean,
-    searchUrl: String,
-    select2Options: String,
-  }
-  
-  // will be reissued as native dom events name prepended with '$' e.g. '$change', '$select2:closing', etc
-  static jQueryEventsToReissue = [ "change", "select2:closing", "select2:close", "select2:opening", "select2:open", "select2:selecting", "select2:select", "select2:unselecting", "select2:unselect", "select2:clearing", "select2:clear" ]
-  
-
-  initialize() {
-    this.dispatchNativeEvent = this.dispatchNativeEvent.bind(this)
-    if (!this.isSelect2LoadedOnWindowJquery) {
-      select2()
-    }
-  }
-
-  get isSelect2LoadedOnWindowJquery() {
-    return window?.$?.fn?.select2 !== undefined
-  }
-
-  get optionsOverride() {
-    if (!this.hasSelect2OptionsValue) { return {} }
-    return this.select2OptionsValue
-  }
 
   connect() {
-    if (this.isSelect2LoadedOnWindowJquery) {
-      this.initPluginInstance()
-    }
+    console.log("tom-select connected")
+    this.initPluginInstance()
   }
 
   disconnect() {
     this.teardownPluginInstance()
   }
 
-  cleanupBeforeInit() {
-    $(this.element).find('.select2-container--default').remove()
-  }
-
   initPluginInstance() {
-    let options = {
-      dropdownParent: $(this.element)
-    };
-
-    if (!this.enableSearchValue) {
-      options.minimumResultsForSearch = -1;
-    }
-
-    options.tags = this.acceptsNewValue
-
-    if (this.searchUrlValue) {
-      options.ajax = {
-        url: this.searchUrlValue,
-        dataType: 'json',
-        // We enable pagination by default here
-        data: function(params) {
-          var query = {
-            search: params.term,
-            page: params.page || 1
-          }
-          return query
-        }
-      }
-    }
-
-    options.templateResult = this.formatState;
-    options.templateSelection = this.formatState;
-    options.width = 'style';
-
-    // Merge in custom options.
-    const custom_options = Object.keys(this.optionsOverride).length > 0 ? JSON.parse(this.optionsOverride) : {}
-    options = {...options, ...custom_options}
-
-    this.cleanupBeforeInit() // in case improperly torn down
-    this.pluginMainEl = this.selectTarget // required because this.selectTarget is unavailable on disconnect()
-    $(this.pluginMainEl).select2(options);
-
-    this.initReissuePluginEventsAsNativeEvents()
+    this.plugin = new TomSelect(this.selectTarget)
   }
 
   teardownPluginInstance() {
```

Comment:
I wonder if `initPluginInstance` and `teardownPluginInstance` add much when they're this simple now. But maybe that's for later!